### PR TITLE
Serve SRI attributes in HTML

### LINKED ISSUE #30: Use SRI for our stylesheet and for any JS
We should probably be using SRI ourselves :)

But let's wait until the front-end is landed and the styles are stable.

### PR DESCRIPTION
**Pull request by neftaly**
Closes #30

### CODE REVIEW

**Comment by neftaly** [comment]
@fmarier 

This is a different approach, using cached run-time generation as opposed to build-time generation.
Unfortunately it's only suitable for backends with state (i.e. Node).

File: templates/index.html [hunk 1]
```diff
@@ -58,8 +58,8 @@
       <p>To fully test your browser for sub-resource integrity support, please open <a href="https://btoews.github.io/sri_test/">this page</a>.</p>
 
       <div class="sri-test" id="sri-badge">
-        <link rel="stylesheet" href="/badge/will-pass.css">
-        <link rel="stylesheet" href="/badge/must-fail.css" integrity="sha256-INVALIDHASH">
+        <link rel="stylesheet" href="/badge/will-pass.css" integrity="type:text/css {{sri "./public/badge/will-pass.css"}}">
```
Comment (fmarier) [inline templates/index.html]: This one we probably want to leave without an integrity attribute just to ensure that the "unsupported" SVG is always displayed.

**Comment by fmarier** [comment]
That's pretty cool, I assume you just released that handlebar-helper-sri module? :)

File: templates/index.html (hunk 1 above)
Comment (neftaly) [inline templates/index.html]: My thoughts too, but what about CSP2 SRI enforcement?

**Comment by neftaly** [comment]
Thanks! I'll add you to the repo, though it's a bit basic.

https://github.com/neftaly/handlebars-helper-sri

File: templates/index.html (hunk 1 above)
Comment (fmarier) [inline templates/index.html]: > My thoughts too, but what about CSP2 SRI enforcement?

That doesn't exist, so we're good :)

Comment (neftaly) [inline templates/index.html]: How about I just remove the "type" directive? There's no plan to enforce content-types in CSP2, right?

Comment (fmarier) [inline templates/index.html]: The `type` _option_ (within the `integrity` attribute) is enforced by SRI. That works in my patch.

Comment (neftaly) [inline templates/index.html]: Sorry, I didn't see your prior comment before writing mine.

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
   },
   "dependencies": {
     "handlebars": "^3.0.0",
+    "handlebars-helper-sri": "0.0.0",
```
Comment (fmarier) [inline package.json]: That's a really weird version number. Why not 0.0.1 at least? :)

File: templates/index.html [hunk 3]
```diff
@@ -59,7 +59,7 @@
 
       <div class="sri-test" id="sri-badge">
         <link rel="stylesheet" href="/badge/will-pass.css">
-        <link rel="stylesheet" href="/badge/must-fail.css" integrity="sha256-INVALIDHASH">
+        <link rel="stylesheet" href="/badge/must-fail.css" integrity="type:invalid/type sha256-INVALIDHASH">
```
Comment (fmarier) [inline templates/index.html]: I see what you mean. Yes, we should only have one failure condition here. No need to include the content-type as well.

File: package.json (hunk 2 above)
Comment (neftaly) [inline package.json]: I usually start at 0.0.0, and wait a week or so to make sure I hadn't overlooked anything :)

File: templates/index.html (hunk 3 above)
Comment (neftaly) [inline templates/index.html]: Ok. If any of this fails in the future, it's on your head :P

**Pull request merged**